letsencrypt/acme/jose/jwk.py

```diff
@@ -0,0 +1,100 @@
+"""JSON Web Key."""
+import binascii
+
+import Crypto.PublicKey.RSA
+
+from letsencrypt.acme.jose import b64
+from letsencrypt.acme.jose import errors
+from letsencrypt.acme.jose import json_util
+
+
+class JWK(json_util.TypedJSONObjectWithFields):
+    # pylint: disable=too-few-public-methods
+    """JSON Web Key."""
+    type_field_name = 'kty'
+    TYPES = {}
+
+
+@JWK.register
+class JWKES(JWK):  # pragma: no cover
+    # pylint: disable=abstract-class-not-used
+    """ES JWK.
+
+    .. warning:: This is not yet implemented!
+
+    """
+    typ = 'ES'
+
+    def fields_to_json(self):
+        raise NotImplementedError()
+
+    @classmethod
+    def fields_from_json(cls, jobj):
+        raise NotImplementedError()
+
+
+@JWK.register
+class JWKOct(JWK):
+    """Symmetric JWK."""
+    typ = 'oct'
+    __slots__ = ('key',)
+
+    def fields_to_json(self):
+        # TODO: An "alg" member SHOULD also be present to identify the
+        # algorithm intended to be used with the key, unless the
+        # application uses another means or convention to determine
+        # the algorithm used.
+        return {'k': self.key}
+
+    @classmethod
+    def fields_from_json(cls, jobj):
+        return cls(key=jobj['k'])
+
+
+@JWK.register
+class JWKRSA(JWK):
+    """RSA JWK."""
+    typ = 'RSA'
+    __slots__ = ('key',)
```

Comment:
`jose.Field` is used for establishing 1-1 relation between JSON object field and internal representation "slot" (attribute): for `JWK` that would be `n = jose.Field('n', ...); e = jose.Field('e', ...)`. It seems more convenient to unpack `n` and `e` into `Crypto.PublicKey.RSA` already at the deserialization step (similarily for the serialization): that is why I override `fields_from_json` and `fields_to_json`. Does this make sense?